recipes/cairo/all/conanfile.py

```diff
@@ -0,0 +1,193 @@
+                    args.append('LIBPNG_PATH=%s' % self.deps_cpp_info['libpng'].rootpath)
+                    args.append('PIXMAN_PATH=%s' % self.deps_cpp_info['pixman'].rootpath)
+                    args.append('FREETYPE_PATH=%s' % self.deps_cpp_info['freetype'].rootpath)
+                    args.append('GOBJECT_PATH=%s' % self.deps_cpp_info['glib'].rootpath)
+                    
+                    env_build.make(args=args)
+                    env_build.make(args=['-C', os.path.join('util', 'cairo-gobject')] + args)
+
+    def _build_configure(self):
+        with tools.chdir(self._source_subfolder):
+            # disable build of test suite
+            tools.replace_in_file(os.path.join('test', 'Makefile.am'), 'noinst_PROGRAMS = cairo-test-suite$(EXEEXT)',
+                                  '')
+            os.makedirs('pkgconfig')
+            for pc_name in  glob.glob('%s/*.pc' % self.build_folder):
+                shutil.copy(pc_name, os.path.join('pkgconfig', os.path.basename(pc_name)))
+            if self.options.enable_ft:
+                tools.replace_in_file(os.path.join(self.source_folder, self._source_subfolder, "src", "cairo-ft-font.c"),
+                                      '#if HAVE_UNISTD_H', '#ifdef HAVE_UNISTD_H')
+
+            pkg_config_path = os.path.abspath('pkgconfig')
+            pkg_config_path = tools.unix_path(pkg_config_path) if self.settings.os == 'Windows' else pkg_config_path
+
+            configure_args = ['--enable-ft' if self.options.enable_ft else '--disable-ft']
+            if self.settings.os != "Windows":
+                configure_args.append('--enable-fc' if self.options.enable_fc else '--disable-fc')
+            if self.settings.os == 'Linux':
+                configure_args.append('--enable-xlib' if self.options.enable_xlib else '--disable-xlib')
+                configure_args.append('--enable-xlib_xrender' if self.options.enable_xlib_xrender else '--disable-xlib_xrender')
+                configure_args.append('--enable-xcb' if self.options.enable_xcb else '--disable-xcb')
+            configure_args.append('--enable-gobject' if self.options.enable_glib else '--disable-gobject')
+            if self.options.shared:
+                configure_args.extend(['--disable-static', '--enable-shared'])
+            else:
+                configure_args.extend(['--enable-static', '--disable-shared'])
+
+            env_build = AutoToolsBuildEnvironment(self)
+            if self.settings.os == 'Macos':
+                env_build.link_flags.extend(['-framework CoreGraphics',
+                                             '-framework CoreFoundation'])
+            if str(self.settings.compiler) in ['gcc', 'clang', 'apple-clang']:
+                env_build.flags.append('-Wno-enum-conversion')
+            with tools.environment_append(env_build.vars):
+                self.run('PKG_CONFIG_PATH=%s NOCONFIGURE=1 ./autogen.sh' % pkg_config_path)
```

Comment:
You need to add a build requirement on `autoconf`, `automake` or `libtool` for this.